### PR TITLE
Add Strong Types for useCookie hook

### LINKED ISSUE #291: add Types
I was looking for a way to add typescript types with using the useCookies hook. What I wanted was a way that the first item in the response array was types with the input array of strings instead of `[name: string]: any` and/or a way to possibly provide a type override if that was not possible so I could add in `useCookies<{myCookie:any}>(['myCookie'])`

### PR DESCRIPTION
**Pull request by amaster507**
Fixes #291

### CODE REVIEW

**Comment by amaster507** [comment]
Type hinting with given dependencies:
image
image
image

Type hinting without given dependencies:
image
image
image

Overriding type hinting (not suggested):
image
image
image

**Comment by amaster507** [comment]
@eXon can I kindly request your review of this PR?

File: packages/react-cookie/src/useCookies.tsx [hunk 1]
```diff
@@ -23,7 +23,7 @@ export default function useCookies(
       const newCookies = cookies.getAll();
 
       if (
-        shouldUpdate(
+        shouldUpdate<T|null,U>(
```
Comment (eXon) [inline packages/react-cookie/src/useCookies.tsx:26]:
```suggestion
        shouldUpdate<T | null, U>(
```

**Review by eXon — COMMENTED** [review]
This looks much better! Just fixing a few whitespaces missing.

**Comment by amaster507** [comment]
Thank you!

**Comment by eXon** [comment]
Released in v4.1.0

**Pull request merged**